lib/task-queue/Queue.js

```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+let Queue = function() {
```

Comment:
I would call this a LIFOQueue. We need to make explicit that this is a stack data structure.